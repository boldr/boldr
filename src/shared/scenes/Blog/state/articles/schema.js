import { schema } from 'normalizr';
import { user } from '../../../../state/modules/users/schema';
import { tag } from '../tags/schema';

const article = new schema.Entity(
  'articles',
  {
    author: user,
    tags: [tag],
  },
  { idAttribute: 'slug' },
);

const arrayOfArticle = new schema.Array(article);

export { article, arrayOfArticle };
