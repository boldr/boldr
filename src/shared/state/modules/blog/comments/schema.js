import { schema } from 'normalizr';
import { user } from '../../users/schema';

const comment = new schema.Entity(
  'comments',
  {
    commenter: user,
  },
  {
    processStrategy: (value, parent, key) => {
      return {
        ...value,
        post: parent.id,
      };
    },
  },
  { idAttribute: 'id' },
);

const arrayOfComment = new schema.Array(comment);

export { comment, arrayOfComment };
