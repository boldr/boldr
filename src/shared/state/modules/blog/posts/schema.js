import { schema } from 'normalizr';
import { user } from '../../users/schema';
import { tag } from '../tags/schema';
import { comment } from '../comments/schema';

const post = new schema.Entity(
  'posts',
  {
    author: user,
    tags: [tag],
    comments: [comment],
  },
  { idAttribute: 'slug' },
);

const arrayOfPost = new schema.Array(post);

export { post, arrayOfPost, comment };
