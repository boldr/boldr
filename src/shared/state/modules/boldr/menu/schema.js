import {normalize, schema} from 'normalizr';

const detail = new schema.Entity('details', {idAttribute: 'id'});
const menu = new schema.Entity(
  'menus',
  {
    details: [detail],
  },
  {idAttribute: 'id'},
);

export {detail, menu};
