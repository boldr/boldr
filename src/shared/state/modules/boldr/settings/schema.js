import { schema } from 'normalizr';

const setting = new schema.Entity('settings', {
  idAttribute: 'key',
});

const arrayOfSetting = new schema.Array(setting);

export { arrayOfSetting, setting };
