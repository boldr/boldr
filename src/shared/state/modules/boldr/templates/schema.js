import {schema} from 'normalizr';

const template = new schema.Entity('templates', {idAttribute: 'id'});
// const tag = new schema.Entity('tags');
const arrayOfTemplate = new schema.Array(template);

export {arrayOfTemplate, template};
