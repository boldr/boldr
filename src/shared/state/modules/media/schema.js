import { schema } from 'normalizr';
import { user } from '../users/schema';

const media = new schema.Entity(
  'media',
  {
    owner: user,
  },
  { idAttribute: 'id' },
);

const arrayOfMedia = new schema.Array(media);

export { media, arrayOfMedia };

/*
id
userId
fileName
safeName
thumbName
safeName
thumbName
 */
