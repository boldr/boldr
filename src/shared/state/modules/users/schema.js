import {schema} from 'normalizr';

const userProcessStrategy = (value, parent, key) => {
  switch (key) {
    case 'author':
      return {
        ...value,
        posts: [parent.id],
      };
    case 'commenter':
      return {
        ...value,
        comments: [parent.id],
      };
    default:
      return {...value};
  }
};

const userMergeStrategy = (entityA, entityB) => {
  return {
    ...entityA,
    ...entityB,
    posts: [...(entityA.posts || []), ...(entityB.posts || [])],
    comments: [...(entityA.comments || []), ...(entityB.comments || [])],
  };
};

const user = new schema.Entity(
  'users',
  {},
  {
    mergeStrategy: userMergeStrategy,
    processStrategy: userProcessStrategy,
    idAttribute: 'id',
  },
);

const arrayOfUsers = [user];

export {user, arrayOfUsers};
