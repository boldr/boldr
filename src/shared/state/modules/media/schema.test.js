import React from 'react';

import {media, arrayOfMedia} from './schema';

describe('media schema', () => {
  it('should match the normalizr-media-schema', () => {
    expect(media).toMatchSnapshot();
  });
});

describe('arrayOfMedia', () => {
  it('should match the normalizr-arrayOfMedia-schema', () => {
    expect(arrayOfMedia).toMatchSnapshot();
  });
});
