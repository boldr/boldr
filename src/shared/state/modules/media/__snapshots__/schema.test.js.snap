// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`arrayOfMedia should match the normalizr-arrayOfMedia-schema 1`] = `
ArraySchema {
  "schema": EntitySchema {
    "_getId": [Function],
    "_idAttribute": "id",
    "_key": "media",
    "_mergeStrategy": [Function],
    "_processStrategy": [Function],
    "schema": Object {
      "owner": EntitySchema {
        "_getId": [Function],
        "_idAttribute": "id",
        "_key": "users",
        "_mergeStrategy": [Function],
        "_processStrategy": [Function],
        "schema": Object {},
      },
    },
  },
}
`;

exports[`media schema should match the normalizr-media-schema 1`] = `
EntitySchema {
  "_getId": [Function],
  "_idAttribute": "id",
  "_key": "media",
  "_mergeStrategy": [Function],
  "_processStrategy": [Function],
  "schema": Object {
    "owner": EntitySchema {
      "_getId": [Function],
      "_idAttribute": "id",
      "_key": "users",
      "_mergeStrategy": [Function],
      "_processStrategy": [Function],
      "schema": Object {},
    },
  },
}
`;
